contents/blog/ab-testing-guide-for-engineers.md

```diff
@@ -0,0 +1,263 @@
+
+It's absolutely **essential** to only log the users in your test who would actually be impacted by your changes. Users who aren't affected by your test should be excluded. If you include unaffected users, their unchanged behavior will mix in with your results, thereby watering down the impact and the clarity of your findings.
+
+To do so, ensure that checking your feature flag and logging their exposure is the *absolute* last condition you check in your code: 
+
+```
+// ❌ Incorrect. Will include unaffected users
+function showNewChanges(user) {
+  if (posthog.getFeatureFlag('experiment-key') === 'control') {
+    return false;
+  }
+
+  if (user.hasCompletedAction) {
+    return false
+  }
+
+  // other checks
+
+  return true
+}
+```
+
+```
+// ✅ Correct. Will exclude unaffected users
+function showNewChanges(user) {
+
+  if (user.hasCompletedAction) {
+    return false
+  }
+
+  // other checks
+
+  if (posthog.getFeatureFlag('experiment-key') === 'control') {
+    return false;
+  }
+
+  return true
+}
+```
+
+## Monitoring your A/B test
+
+Once you've started your A/B test, it's a good idea to check in on it 24-48 hours after launch to ensure that everything is running correctly. Here's a list of things to check:
+
+* Check your exposure logging. For example, that the volume of users assigned to the control and test variant is what you'd expect it to be.
+
+* Check your event logging – i.e. that you're receiving all the events you expect to receive, and in the right ratios.
+
+* Monitor crashes or error rates to ensure that your test is not causing any issues. 
+
+Once you're sure that everything is running smoothly, try to avoid frequently checking the results until the test is finished. Checking too often can lead to reactive decisions based on incomplete data. It's usually best to wait until the test ends to make any decisions, as you'll have the best data at that point.
+
+## Analyzing your results
+
+Once your test has run its duration, and you've collected enough data, you're ready to look at the results. What you're looking for is statistical significant change in any of the following metrics:
+
+1. **Your goal metric.**
+
+2. **Any secondary metrics.** These are usually metrics that are closely related to your goal metric. For example, if your goal metric is to increase paid conversions, a secondary metric may be to increase sign-ups.
+
+3. **Any counter metrics.** These are metrics that ensure your user experience isn't deteriorating. For example, if your goal is to increase sign-ups, you should also verify your churn rate is not increasing.
+
+Most A/B testing tools, including [PostHog](/ab-testing/features), will calculate statistical significance for you. To do so manually, you'll need to input the number of users in each variant, and the conversion rate for each metric, into an online calculator.
```

Comment:
```suggestion
These metrics must have statistically significant changes because that means the impact is larger than random in 95% (or your confidence level) of cases. Most A/B testing tools, including [PostHog](/ab-testing), calculate statistical significance for you. To do so manually, you'll need to input the number of users in each variant, and the conversion rate for each metric, into an online calculator.
```

Reply:
I'm not sure if this is more or less confusing. It just might be useful to note why we care about statistical significant. Feel free to trash if it makes it worse.